ui/src/app/utils/leo-runtime-initializer.spec.tsx

```diff
@@ -163,6 +163,31 @@ describe('RuntimeInitializer', () => {
     expect(runtime.status).toEqual(RuntimeStatus.Running);
   });
 
+  it('should lazily create user\'s most runtime if a valid one exists', async() => {
+    serverConfigStore.next({gsuiteDomain: 'researchallofus.org', enableCustomRuntimes: true});
+    mockGetRuntime.mockImplementation(namespace => {
+      return {
+        ...defaultRuntime(),
+        gceConfig: {
+          diskSize: 777,
+          machineType: 'n1-standard-16'
+        },
+        status: RuntimeStatus.Deleted
+      }; });
+
+    LeoRuntimeInitializer.initialize({
+      workspaceNamespace: workspaceNamespace,
+    });
+    await new Promise(setImmediate);
```

Comment:
I'm pretty sure this does what I want but I don't really understand how it works. Can someone explain to me why this works but `await LeoRuntimeInitializer.....` does not?

Reply:
We're using fake timers in this test. The LeoRuntimeInitializer uses `setTimeout()` to do the polling, which means it won't terminate unless we tick the timers. see the helper function `runInitializerAndTimers` which handles this properly, you may want to use that here.

Reply:
Also, as to why this `await new Promise(setImmediate)` appeared to work: what this does is allow the call stack to resolve for any calbacks/promises that are pending. In this case, I expect this gives the first call to `getRuntime` in the initializer a chance to return and be processed, which may be enough to satisfy your test constraints.

Reply:
I see. thanks for the explanation. I actually did start out by using `runInitializerAndTimers` but I only wanted it to resolve once and not make the subsequent calls to `poll` which is why I was looking for an alternative.

Reply:
ack, you can also do `await waitForTick` with is effectively achieving the same thing and is more standard to use in our tests

Reply:
the existing functions look like they take in a `wrapper` component but I don't think we actually have one in these tests